icecreamshop/src/main/java/com/summerschool/icecreamshop/service/RateService.java

```diff
@@ -14,9 +14,17 @@
 @Service
 public class RateService {
 
-    @Autowired
     private RateRepository rateRepository;
 
+    @Autowired
+    public RateService(RateRepository rateRepository) {
+        this.rateRepository = rateRepository;
+    }
+
+    public Rate add(Rate rate) {
+        return rateRepository.save(rate);
```

Comment:
cred ca rateRepository.save poate arunca o exceptie in cazul in care salvarea nu se poate realiza.  ai putea sa adaugi un try {return rateRepository.save(rate)} catch(Exception) { logError ...  return null}